pkg/envoy/lds/http_connection_test.go

```diff
@@ -0,0 +1,135 @@
+package lds
+
+import (
+	"testing"
+	"time"
+
+	xds_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/openservicemesh/osm/pkg/auth"
+)
+
+func TestHTTPConnBuildFromOptions(t *testing.T) {
+	notContains := func(filters []*xds_hcm.HttpFilter, filterName string) bool {
+		for _, f := range filters {
+			if f.Name == filterName {
+				return false
+			}
+		}
+		return true
+	}
+	contains := func(filters []*xds_hcm.HttpFilter, filterName string) bool {
+		return !notContains(filters, filterName)
+	}
+
+	testCases := []struct {
+		name       string
+		option     httpConnManagerOptions
+		assertFunc func(*assert.Assertions, *xds_hcm.HttpConnectionManager, error)
```

Comment:
I notice the `error` is never checked in any test. Since it looks like any error would ultimately come from `ptypes.MarshalAny` (which I haven't found a way to reliably make fail), I'd be in favor of removing it here and adding a `assert.NoError(err)` directly in the loop over the tests where we call `buildFromOptions()`.